src/proc/validator.rs

```diff
@@ -228,14 +228,18 @@ impl Validator {
                     | (crate::ShaderStage::Fragment, crate::StorageClass::Input) => {
                         match module.types[var.ty].inner {
                             crate::TypeInner::Scalar { kind, .. }
-                            | crate::TypeInner::Vector { kind, .. }
-                            | crate::TypeInner::Matrix { kind, .. } => {
+                            | crate::TypeInner::Vector { kind, .. } => {
                                 if kind != crate::ScalarKind::Float
                                     && var.interpolation != Some(crate::Interpolation::Flat)
                                 {
                                     return Err(EntryPointError::InvalidIntegerInterpolation);
                                 }
                             }
+                            crate::TypeInner::Matrix { .. } => {
+                                if var.interpolation != Some(crate::Interpolation::Flat) {
```

Comment:
this condition should be removed from here, the code above checked "kind != crate::ScalarKind::Float &&  this", so in this branch it's guaranteed false

Reply:
I'm sorry, but I have no idea what interpolation means. So I can remove this entire match arm? As this is the only check.

Reply:
This would essentially be `crate::TypeInner::Matrix { .. } => {}`